### PR TITLE
Add abandoned mine lands data

### LINKED ISSUE #1780: Add eAMLIS data to tool
Add eAMLIS to legacy pollution, including streamlining the points -> tracts code

### PR DESCRIPTION
**Pull request by mattbowen-usds**
Closes #1780

### CODE REVIEW

File: data/data-pipeline/data_pipeline/etl/score/config/datasets.yml [hunk 1]
```diff
@@ -145,6 +145,18 @@ datasets:
         field_type: bool
         include_in_tiles: false
         include_in_downloadable_files: false
+  - long_name: "Abandoned Mine Land Inventory System"
+    short_name: "eAMLIS"
+    module_name: "eamlis"
+    load_fields:
+      - short_name: "has_amlis"
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/score/config/datasets.yml:152]: nit: "has_aml"

File: data/data-pipeline/data_pipeline/etl/score/config/datasets.yml [hunk 2]
```diff
@@ -145,6 +145,18 @@ datasets:
...
+        df_field_name: "BINARY_HAS_AMLIS"
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/score/config/datasets.yml:153]: nit: "AML_BOOLEAN"

File: data/data-pipeline/data_pipeline/etl/score/config/datasets.yml [hunk 3]
```diff
@@ -145,6 +145,18 @@ datasets:
...
+        long_name: "Is there at least one abandoned mine in this census tract?"
+        description_short: 
+          "Whether the tract has an abandoned mine"
+        field_type: bool
+        include_in_tiles: false
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/score/config/datasets.yml:158]: I think these should both be true? Because it comes into the score. As far as I can tell, this doesn't currently matter, though

Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/score/config/datasets.yml:158]: Cool, will change, and thank you.

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+from data_pipeline.config import settings
+
+from data_pipeline.etl.base import ExtractTransformLoad, ValidGeoLevel
+from data_pipeline.etl.sources.geo_utils import add_tracts_for_geometries
+from data_pipeline.utils import get_module_logger
+
+logger = get_module_logger(__name__)
+
+
+class AbandonedMineLandInventorySystem(ExtractTransformLoad):
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:13]: I think our convention has been "AbandonedMineETL" or something similar, but I don't have strong feelings?

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    """Data from Office Of Surface Mining Reclamation and Enforcement's
+    eAMLIS. These are the locations of abandoned mines.
+    """
+
+    # Metadata for the baseclass
+    NAME = "eamlis"
+    GEO_LEVEL = ValidGeoLevel.CENSUS_TRACT
+    BINARY_HAS_AMLIS: str = "Has abandoned mine"
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:21]: why is this here rather than in the yml file?

Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:21]: Oh good catch! Thank you --- I had put it in before I wrote the YAML when I was deciding what vars I needed. THank you!

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 4 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:13]: Thank you!

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        )
+        gdf = gpd.GeoDataFrame(
+            df,
+            geometry=gpd.points_from_xy(
+                x=df["Longitude"],
+                y=df["Latitude"],
+            ),
+            crs="epsg:4326",
+        )
+        gdf.drop_duplicates(subset=["geometry"], inplace=True, keep="last")
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:58]: `in_place = True` can be dangerous. Can we reassign to the same frame?

Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:58]: Oh oh how so!? and of course!

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        gdf_tracts = add_tracts_for_geometries(gdf)
+        gdf_tracts.drop_duplicates(self.GEOID_TRACT_FIELD_NAME, inplace=True)
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:60]: Huh, I wonder why there are duplicate tract_ids here. I haven't worked a lot with this data, but this makes me kind of nervous. Do you have a sense of what's going on?

**Review by emma-nechamkin — APPROVED** [review]

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 6 above)
Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:58]: `gdf = gdf.drop_duplicates(subset=["geometry"], keep="last")` :)

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 7 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py:60]: Yeah! If you have two mines in the same tract, that tract would show up repeatedly in the data set. So like this happens in Alaska, for example. AK000096 and AK000080 are really close to one another and so tract 02261000100 shows up repeatedly.

File: data/data-pipeline/data_pipeline/etl/score/config/datasets.yml (hunk 1 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/score/config/datasets.yml]: Good call -- done in 702804e39047697fd3673a64fdd56256dfd82d58

File: data/data-pipeline/data_pipeline/etl/score/config/datasets.yml (hunk 2 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/score/config/datasets.yml]: Done in 702804e39047697fd3673a64fdd56256dfd82d58

File: data/data-pipeline/data_pipeline/etl/score/config/datasets.yml (hunk 3 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/score/config/datasets.yml]: Thank you! I always feel like I'm guessing at these --- done in 702804e39047697fd3673a64fdd56256dfd82d58

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 4 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py]: That is a less silly name: 702804e39047697fd3673a64fdd56256dfd82d58

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 5 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py]: Fixed in 702804e39047697fd3673a64fdd56256dfd82d58

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 7 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py]: We talked in slack and we are cool!

File: data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py (hunk 6 above)
Comment (mattbowen-usds) [reply data/data-pipeline/data_pipeline/etl/sources/eamlis/etl.py]: Oh no I meant how's it dangerous --- like I have been burned by so many things but not inplace.
